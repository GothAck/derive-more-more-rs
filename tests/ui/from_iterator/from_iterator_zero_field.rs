use derive_more_more::FromIterator;

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct NamedZero {}

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct UnnamedZero();

#[derive(Default, FromIterator)]
#[from_iterator(usize)]
struct UnitZero;

#[derive(Default, FromIterator)]
struct NamedZeroNoAttr {}

#[derive(Default, FromIterator)]
struct UnnamedZeroNoAttr();

#[derive(Default, FromIterator)]
struct UnitZeroNoAttr;

fn main() {}
