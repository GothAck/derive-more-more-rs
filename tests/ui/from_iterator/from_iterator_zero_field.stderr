error: Struct has no fields
 --> tests/ui/from_iterator/from_iterator_zero_field.rs:4:1
  |
4 | #[from_iterator(usize)]
  | ^

error: Struct has no fields
 --> tests/ui/from_iterator/from_iterator_zero_field.rs:8:1
  |
8 | #[from_iterator(usize)]
  | ^

error: Struct has no fields
  --> tests/ui/from_iterator/from_iterator_zero_field.rs:12:1
   |
12 | #[from_iterator(usize)]
   | ^

error: Struct has no fields
  --> tests/ui/from_iterator/from_iterator_zero_field.rs:16:1
   |
16 | struct NamedZeroNoAttr {}
   | ^^^^^^

error: Struct has no fields
  --> tests/ui/from_iterator/from_iterator_zero_field.rs:19:1
   |
19 | struct UnnamedZeroNoAttr();
   | ^^^^^^

error: Struct has no fields
  --> tests/ui/from_iterator/from_iterator_zero_field.rs:22:1
   |
22 | struct UnitZeroNoAttr;
   | ^^^^^^
